### PR TITLE
[FSSDK-9382] switch client init args from positional to keyword

### PR DESCRIPTION
**Pull request by andrewleap-optimizely**
## Summary
The Optimizely constructor uses positional args and since all of them are optional, it can become unwieldy.

This PR changes the positional args to keyword.

## Ticket
[FSSDK-9382](https://jira.sso.episerver.net/browse/FSSDK-9382)

## Issues
https://github.com/optimizely/ruby-sdk/issues/243

### CODE REVIEW

File: spec/project_spec.rb [hunk 1]
```diff
@@ -48,7 +48,7 @@
   let(:version) { Optimizely::VERSION }
   let(:impression_log_url) { 'https://logx.optimizely.com/v1/events' }
   let(:conversion_log_url) { 'https://logx.optimizely.com/v1/events' }
-  let(:project_instance) { Optimizely::Project.new(config_body_JSON, nil, spy_logger, error_handler, false, nil, nil, nil, nil, nil, [], {batch_size: 1}) }
+  let(:project_instance) { Optimizely::Project.new(datafile: config_body_JSON, logger: spy_logger, error_handler: error_handler, event_processor_options: {batch_size: 1}) }
```
Comment (mikechu-optimizely) [inline spec/project_spec.rb:51]: Do you want to change `error_handler: error_handler,` to `error_handler:,`?

Comment (mikechu-optimizely) [reply spec/project_spec.rb:51]: ...and in other spots too?  Is the linter looking for this in the SDK like the testapp?

**Review by mikechu-optimizely — CHANGES_REQUESTED** [review]
Some thoughts and suggestions for whatever my Ruby review's worth.

File: spec/project_spec.rb (hunk 1 above)
Comment (andrewleap-optimizely) [reply spec/project_spec.rb:51]: We still support 3.0 and the shorthand syntax was introduced in 3.1

**Pull request merged**